File: ASPNet/MeuProjetoCrudAPI/Models/ret-taskModel.cs
using System.Runtime.Serialization;

namespace MeuProjetoCrudAPI.Models
{
    [DataContract]
    public class ret_taskModel : ReturnModel<List<TaskModel>>
    {
        [DataMember(Name = "Data")]
        public override List<TaskModel> Data { get; set; }

        public ret_taskModel()
        {
            this.Error = false;
            this.ErrorMessage = "";
            this.Data = new List<TaskModel>();
        }
        
    }
}

File: Angular/MeuProjetoCrudAngular/src/app/components/tasks/Models/ret-taskModel.ts
import { TaskModel } from './TaskModel';

export class RetTaskModel {
  error: boolean;
  errorMessage: string;
  data: TaskModel[];

  constructor(
    error: boolean = false,
    errorMessage: string = '',
    data: TaskModel[] = []
  ) {
    this.error = error;
    this.errorMessage = errorMessage;
    this.data = data;
  }
}
